Development/nmos/sdp_utils.h

```diff
@@ -245,11 +245,41 @@ namespace nmos
             {}
         } mediaclk;
 
+        // RTP Header Extensions
+        // See https://www.rfc-editor.org/rfc/rfc5285#section-5
+        struct extmap_t
+        {
+            uint64_t local_id;
+            sdp::direction direction;
+            utility::string_t uri;
+            utility::string_t ext_attributes;
+
+            extmap_t() : local_id() {}
+            extmap_t(uint64_t local_id, const utility::string_t& uri): local_id(local_id), uri(uri) {}
+            extmap_t(uint64_t local_id, const sdp::direction& direction, const utility::string_t& uri): local_id(local_id), direction(direction), uri(uri) {}
+            extmap_t(uint64_t local_id, const utility::string_t& uri, const utility::string_t& ext_attributes): local_id(local_id), uri(uri), ext_attributes(ext_attributes) {}
+            extmap_t(uint64_t local_id, const sdp::direction& direction, const utility::string_t& uri, const utility::string_t& ext_attributes): local_id(local_id), direction(direction), uri(uri), ext_attributes(ext_attributes) {}
+        };
+        std::vector<sdp_parameters::extmap_t> extmap;
```

Comment:
I think `extmap` is a media attribute not a session attribute, right?
And multiple `extmap` attributes may be needed for a media description?
But also we have ST 2022-7, so do we need different set of `extmap` for each leg?

Reply:
https://www.rfc-editor.org/rfc/rfc8285#section-5 says
> The mapping MAY be provided per media stream (in the media-level section(s) of SDP, i.e., after an "m=" line) or globally for all streams (i.e., before the first "m=" line, at session level). The definitions MUST be either all session level or all media level; it is not permitted to mix the two styles. In addition, as noted above, the IDs used MUST be unique in each media section of the SDP or unique in the session for session-level SDP declarations.

If I correctly understand ST 2022-7, it's about two same RTP streams going different paths (different UDP headers). In this case, RTP headers are the same and it still makes sense to put `extmap` at session level.

Reply:
Ah, interesting! OK then...

Reply:
Actually... having looked back at examples of `extmap` all the ones I find use the media-level attributes 😕

https://github.com/search?q=org%3AAMWA-TV+extmap&type=code

Reply:
Resolved by delegating this decision to application code in the latest refactor to add _nmos/sdp_attributes.h_.